### PR TITLE
feat: mark completed ui selector

### PR DESCRIPTION
**Pull request by sjaanus**
image

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/MarkCompletedDialogue.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    isOpen: boolean;
+    setIsOpen: (open: boolean) => void;
+    onComplete: () => void;
+    projectId: string;
+    featureId: string;
+}
+
+type Status = 'kept' | 'discarded' | 'kept-with-variant';
+
+export const MarkCompletedDialogue = ({
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/MarkCompletedDialogue.tsx:19]: There's a bunch of state in this component that deserves a test (separate PR is fine)

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/SingleVariantOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    />
+                    {option}
+                </li>
+            )}
+            renderInput={(params) => (
+                <TextField {...params} placeholder='Select variant' />
+            )}
+            fullWidth
+            onChange={(_, selectedValue) => {
+                onSelect(selectedValue as string);
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/SingleVariantOptions.tsx:40]: String(selectedValue) would work w/o type assertions

**Review by kwasniew — APPROVED** [review]
Approving but I'd consider extracting dialogue similar to archive dialogue. Both in single feature view and search view.

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/FeatureLifecycle.tsx [hunk 3]
```diff
@@ -22,30 +23,35 @@ export const FeatureLifecycle: FC<{
...
+            <FeatureLifecycleTooltip
+                stage={currentStage!}
+                onArchive={onArchive}
+                onComplete={() => setShowMarkCompletedDialogue(true)}
+                onUncomplete={onUncompleteHandler}
+                loading={loading}
+            >
+                <FeatureLifecycleStageIcon stage={currentStage!} />
+            </FeatureLifecycleTooltip>
+            <MarkCompletedDialogue
```
[the remark this answers is not in the document]
Comment (sjaanus) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureLifecycle/FeatureLifecycle.tsx]: Moved out

**Pull request merged**